src/sas/qtgui/Plotting/Plotter.py

```diff
@@ -448,32 +465,97 @@ def onRemoveText(self):
+        """
+        self.toolbar.release_zoom(event)
+        self._zoom_pan_handler(event)
+
+    def _pan_handler(self, event):
+        """
+        Explicitly call toolbar method to ensure range is changed. In MPL 2.2, local events take precedence over the
+        toolbar events, so the range isn't panned until after _zoom_pan_handler is run.
+        """
+        self.toolbar.release_pan(event)
+        self._zoom_pan_handler(event)
+
+    def _home(self, event):
+        """
+        Catch home button click events
+        """
+        self.onResetGraphRange()
+
+    def _back(self, event):
+        """
+        Catch back button click events
+        """
+        self.toolbar.back()
+        self._zoom_pan_handler(event)
+
+    def _forward(self, event):
+        """
+        Catch forward button click events
+        """
+        self.toolbar.forward()
+        self._zoom_pan_handler(event)
+
+    def _pan(self, event):
+        """
+        Catch pan button click events
+        """
+        self.canvas.mpl_connect('button_release_event', self._pan_handler)
+
+    def _zoom(self, event):
+        """
+        Catch zoom button click events
+        """
+        self.canvas.mpl_connect('button_release_event', self._zoom_handler)
+
     def onSetGraphRange(self):
         """
         Show a dialog allowing setting the chart ranges
         """
         # min and max of data
         if self.setRange.exec_() == QtWidgets.QDialog.Accepted:
-            x_range = self.setRange.xrange()
-            y_range = self.setRange.yrange()
-            self.setRange.rangeModified = True
-            if x_range is not None and y_range is not None:
-                self.ax.set_xlim(x_range)
-                self.ax.set_ylim(y_range)
-                self.canvas.draw_idle()
+            self._setGraphRange()
 
-    def onResetGraphRange(self):
-        """
-        Resets the chart X and Y ranges to their original values
-        """
-        for d in self.data:
-            x_range = (d.x.min(), d.x.max())
-            y_range = (d.y.min(), d.y.max())
+    def _setGraphRange(self):
+        x_range = self.setRange.xrange()
+        y_range = self.setRange.yrange()
         if x_range is not None and y_range is not None:
+            self.setRange.rangeModified = (self.setRange.defaultXRange != x_range
+                                           or self.setRange.defaultYRange != y_range)
             self.ax.set_xlim(x_range)
             self.ax.set_ylim(y_range)
             self.canvas.draw_idle()
 
+    def onResetGraphRange(self):
+        """
+        Resets the chart X and Y ranges to their original values
+        """
+        # Clear graph and plot everything again
+        mpl.pyplot.cla()
+        self.ax.cla()
+        self.setRange = None
+        for ids in self.plot_dict:
+            self.plot(data=self.plot_dict[ids], hide_error=self.plot_dict[ids].hide_error)
+
```

Comment:
`Plot` accepts more arguments:
```
def plot(self, data=None, color=None, marker=None, hide_error=False, transform=True):
```
If your plot has non-default marker or color this probably will reset those properties.

Reply:
The markers and colors are stored in each data set held within `self.plot_dict`. If either `color | marker` is not passed to `self.plot`, the respective value is taken from the passed data set. I'd have to do the same here, so I think it would be redundant.